lib/email-addresses.js

```diff
@@ -822,9 +853,23 @@ function parse5322(opts) {
                 delete ret[i].node;
             }
         }
-        return ret;
+        return ret;        
     }
-
+    
+    function addrListGiveResult(ast) {
+        return _giveResult(ast, ['group', 'mailbox']);
```

Comment:
No, I can clean that up now. It happened because they actually have different definitions, but it turns out what you *want* from the results for each is achieved the same way. I started out with different calls to `_giveResults()`. They ended up identical when I carefully examined the grammar and realized that everything boils down to `address`, which is `mailbox / group` anyway.